### PR TITLE
VIDEO-9998 : implement bandwidth profile option- maxSwitchedOnTracks

### PR DESCRIPTION
**Pull request by makarandp0**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### Description
This PR adds new bandwidth profile connect option `maxSwitchedOnTracks` that can be used by client to control max number of tracks received for both audio and video. This option is mainly to make it easy to test large group rooms, we have not decided yet if to make these options available in final API.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### CODE REVIEW

**Comment by charliesantos** [comment]
@makarandp0 do we have an IDL/FRD/Blueprint section that describes this feature? Or all details are in this PR? I understand this feature is not final.

**Comment by makarandp0** [comment]
@charliesantos, yes we have a these new flags mentioned in [blueprint](https://docs.google.com/document/d/1Utox0H6ygX7v4K0_-tOIPi9amIZgiccvW05Jg-Fuuaw/edit) and [idl](https://code.hq.twilio.com/client/video-sdk-api/pull/96) here.

File: lib/connect.js [hunk 1]
```diff
@@ -354,6 +354,11 @@ function connect(token, options) {
         options.clientTrackSwitchOffControl = 'auto';
       }
 
+      if ('maxTracks' in options.bandwidthProfile.video) {
+        // move maxTracks over to new property that will be used to create RSP payload
+        options.bandwidthProfile.video.maxSwitchedOnTracks = options.bandwidthProfile.video.maxTracks;
```
Comment (charliesantos) [inline lib/connect.js:359]: Aren't we removing maxTracks anyway so we don't need this check?

File: lib/util/validate.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ function validateBandwidthProfile(bandwidthProfile) {
     { prop: 'dominantSpeakerPriority', values: Object.values(trackPriority) },
     { prop: 'maxSubscriptionBitrate', type: 'number' },
     { prop: 'maxTracks', type: 'number' },
```
Comment (charliesantos) [inline lib/util/validate.js:20]: Is it safe to remove this line?

File: lib/connect.js (hunk 1 above)
Comment (makarandp0) [reply lib/connect.js:359]: `maxTracks` is marked as deprecated for now. But yes we do need to stop supporting this and  few other properties and remove related code. but that can come later.

File: lib/util/index.js [hunk 3]
```diff
@@ -671,7 +703,7 @@ function createBandwidthProfileVideoPayload(bandwidthProfileVideo) {
   return createRSPPayload(bandwidthProfileVideo, [
     { prop: 'dominantSpeakerPriority', type: 'string', payloadProp: 'active_speaker_priority' },
     { prop: 'maxSubscriptionBitrate', type: 'number', payloadProp: 'max_subscription_bandwidth' },
-    { prop: 'maxTracks', type: 'number', payloadProp: 'max_tracks' },
+    { prop: 'maxSwitchedOnTracks', type: 'number', payloadProp: 'max_tracks' },
```
Comment (charliesantos) [inline lib/util/index.js:706]: So RSP is not updated yet to stop using max_tracks?

**Review by charliesantos — APPROVED** [review]

**Pull request merged**